src/core_ocean/shared/mpas_ocn_gm.F

```diff
@@ -486,86 +488,93 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, diagnosticsPool, &
+                  h2 = layerThicknessEdge(k,iEdge)
+                  ! Using second-order interpolation below
+                  gradDensityTopOfEdge(k,iEdge) = (h2 * gradDensityEdge(k-1,iEdge) + h1 * &
+                                gradDensityEdge(k,iEdge)) / (h1 + h2)
+                  gradZMidTopOfEdge(k,iEdge) = (h2 * gradZMidEdge(k-1,iEdge) + h1 * &
+                                gradZMidEdge(k,iEdge)) / (h1 + h2)
+               end do
 
-            ! Approximation of values on the top and bottom interfaces through the idea of having ghost cells above
-            ! the top and below the bottom layers of the same depths and density.
-            gradDensityTopOfEdge(1,iEdge) = gradDensityEdge(1,iEdge)
-            gradDensityTopOfEdge(maxLevelEdgeTop(iEdge)+1,iEdge) = gradDensityEdge(maxLevelEdgeTop(iEdge),iEdge)
-            gradZMidTopOfEdge(1,iEdge) = gradZMidEdge(1,iEdge)
-            gradZMidTopOfEdge(maxLevelEdgeTop(iEdge)+1,iEdge) = gradZMidEdge(maxLevelEdgeTop(iEdge),iEdge)
-         end if
-      end do
-      !$omp end do
+               ! Approximation of values on the top and bottom interfaces through the idea of having ghost cells 
+               ! above the top and below the bottom layers of the same depths and density.
+               gradDensityTopOfEdge(1,iEdge) = gradDensityEdge(1,iEdge)
+               gradDensityTopOfEdge(maxLevelEdgeTop(iEdge)+1,iEdge) = gradDensityEdge(maxLevelEdgeTop(iEdge),iEdge)
+               gradZMidTopOfEdge(1,iEdge) = gradZMidEdge(1,iEdge)
+               gradZMidTopOfEdge(maxLevelEdgeTop(iEdge)+1,iEdge) = gradZMidEdge(maxLevelEdgeTop(iEdge),iEdge)
+            end if
+         end do
+         !$omp end do
 
-      !$omp do schedule(runtime) private(k)
-      do iEdge = 1, nEdges
-         if (maxLevelEdgeTop(iEdge) .GE. 1) then
-            do k = 1, maxLevelEdgeTop(iEdge)+1
-               gradDensityConstZTopOfEdge(k,iEdge) = gradDensityTopOfEdge(k,iEdge) - dDensityDzTopOfEdge(k,iEdge) &
+         !$omp do schedule(runtime) private(k)
+         do iEdge = 1, nEdges
+            if (maxLevelEdgeTop(iEdge) .GE. 1) then
+               do k = 1, maxLevelEdgeTop(iEdge)+1
+                  gradDensityConstZTopOfEdge(k,iEdge) = gradDensityTopOfEdge(k,iEdge) - dDensityDzTopOfEdge(k,iEdge) &
                                                    * gradZMidTopOfEdge(k,iEdge)
-            end do
-         end if
-      end do
-      !$omp end do
-      !$omp end parallel
+               end do
+            end if
+         end do
+         !$omp end do
+         !$omp end parallel
 
+         nEdges = nEdgesArray(3)
          ! For config_GM_closure = 'N2_dependent' use a scaling to taper gmBolusKappa
          ! based on stratification relative to the maximum in the column
          if (local_config_GM_kappa_lat_depth_variable) then
             !$omp parallel
             !$omp do schedule(runtime) private(k, BruntVaisalaFreqTopEdge, maxN)
```

Comment:
This OpenMP directive will need `cell1` and `cell2` added to the `private()` clause.